packages/relay/src/RelayServer.ts

```diff
@@ -823,6 +809,57 @@ latestBlock timestamp   | ${latestBlock.timestamp}
     return getLatestEventData(events)
   }
 
+  async _resolveWithdrawalConfig (filename: string): Promise<{ withdrawalAmount: BN, repeat: boolean }> {
+    this.logger.debug('Parsing withdrawal config file')
+    const withdrawConfig: { repeat: boolean, withdrawOnEthBalanceReached: number, leaveManagerWithAmountEth: number, smallAmount: boolean } = JSON.parse(
+      fs.readFileSync(filename, 'utf8'))
+    ow(withdrawConfig, ow.object.exactShape(WithdrawalConfigShape))
+    const withdrawOnEthBalanceReached = toBN(withdrawConfig.withdrawOnEthBalanceReached)
+    const leaveManagerWithAmountEth = toBN(withdrawConfig.leaveManagerWithAmountEth)
+    if (withdrawOnEthBalanceReached.lt(leaveManagerWithAmountEth)) {
+      throw new Error('withdrawOnEthBalanceReached must be at least leaveManagerWithAmountEth')
+    }
+    if (leaveManagerWithAmountEth.lt(toBN(this.config.managerTargetBalance))) {
+      throw new Error(`leaveManagerWithAmountEth must be at least managerTargetBalance ${this.config.managerTargetBalance}`)
+    }
+
+    if (withdrawOnEthBalanceReached.sub(leaveManagerWithAmountEth).lt(toBN(1e18)) && !withdrawConfig.smallAmount) {
+      throw new Error('Are you really sure that you want to withdraw less than 1 eth?? Add "smallAmount": true to your config then')
+    }
+    const managerHubBalance = await this.relayHubContract.balanceOf(this.managerAddress)
+    let withdrawalAmount = managerHubBalance.sub(leaveManagerWithAmountEth)
+    if (withdrawOnEthBalanceReached.gt(managerHubBalance)) {
+      // We didn't reach the given hub eth balance to withdraw yet
+      withdrawalAmount = toBN(0)
+    }
+    return { withdrawalAmount, repeat: withdrawConfig.repeat }
+  }
+
+  async withdrawToOwnerIfNeeded (blockNumber: number): Promise<PrefixedHexString[]> {
+    try {
+      let txHashes: PrefixedHexString[] = []
+      const filename = `${this.config.workdir}/withdraw.json`
+      if (!this.isReady() || !fs.existsSync(filename)) {
+        return txHashes
+      }
+      const { withdrawalAmount, repeat } = await this._resolveWithdrawalConfig(filename)
+      if (withdrawalAmount.eqn(0)) {
+        return txHashes
+      }
+      txHashes = txHashes.concat(await this.registrationManager._sendManagerHubBalanceToOwner(blockNumber, withdrawalAmount))
+      if (!repeat) {
+        this.logger.info('Removing withdraw file.')
+        fs.rmSync(filename)
```

Comment:
maybe instead of removing the file, rename it to filename."done" ? this way it can be repeated by moving it back (or, if it failed - check the content of the file to understand why)
if that ".done" file already exist, it should be removed first. no need for more than one backup..

Reply:
Right.